supervisor/src/bin/supervisor-encore.rs

```diff
@@ -0,0 +1,278 @@
+use crate::encore::runtime::v1 as runtimepb;
+use base64::Engine;
+use encore_supervisor::supervisor::{Process, Supervisor};
+use prost::Message;
+use std::fs::File;
+use std::io::Read;
+use std::{collections::HashMap, env};
+use tokio_util::sync::CancellationToken;
+
+pub mod encore {
+    pub mod runtime {
+        pub mod v1 {
+            include!(concat!(env!("OUT_DIR"), "/encore.runtime.v1.rs"));
+        }
+    }
+}
+
+// Supervisor config is the config bundled with the supervisor binary
+// It contains the list of available binaries and which services and gateways they implement
+#[derive(serde::Serialize, serde::Deserialize)]
+struct SupervisorConfig {
+    procs: Vec<Proc>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct Proc {
+    id: String,
+    command: Vec<String>,
+    env: Vec<String>,
+    services: Vec<String>,
+    gateways: Vec<String>,
+}
+
+// Process config is the config for a given process
+// It overrides the RuntimeConfig for local service discovery, port allocation
+// and which services and gateways are hosted on this process
+#[derive(serde::Serialize, serde::Deserialize)]
+struct ProcessConfig {
+    local_service_ports: HashMap<String, u16>,
+    hosted_gateways: Vec<String>,
+    hosted_services: Vec<String>,
+}
+
+// RuntimeConfig is a partial version of the config used for GO apps
+// Only hosted services and gateways are parsed as those are the only ones
+// we need to produce ProcessConfigs
+#[derive(serde::Serialize, serde::Deserialize)]
+struct RuntimeConfig {
+    #[serde(default)]
+    hosted_services: Vec<String>,
+    #[serde(default)]
+    gateways: Vec<GatewayConfig>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct GatewayConfig {
+    name: String,
+}
+
+// Create a process config for a given set of services and gateways
+fn create_process_config(
+    services: Vec<String>,
+    gateways: Vec<String>,
+    port: u16,
+    service_ports: &HashMap<String, u16>,
+    cfg: &SupervisorConfig,
+) -> Result<Process, String> {
+    // Append all supervisor environment variables
+    let mut env = std::env::vars().collect::<HashMap<String, String>>();
+
+    // Find a process config that contains all the services and gateways
+    let binary_config = cfg
+        .procs
+        .iter()
+        .find(|p| {
+            (services.iter().all(|s| p.services.contains(s)))
+                && (gateways.iter().all(|g| p.gateways.contains(g)))
+        })
+        .ok_or_else(|| {
+            format!(
+                "No matching proc found for services {:?} gateways {:?}",
+                services, gateways
+            )
+        })?;
+
+    // Add proc-specific environment variables
+    env.extend(binary_config.env.iter().map(|e| {
+        let parts: Vec<&str> = e.splitn(2, '=').collect();
+        (
+            parts[0].to_string(),
+            parts.get(1).map_or("".to_string(), |&v| v.to_string()),
```

Comment:
tiny nit to save some characters

```suggestion
            parts.get(1).unwrap_or(&"").to_string(),
```
alternatively 
```
parts.get(1).map_or("", |v| v).to_string();
```